src/yunohost/dyndns.py

```diff
@@ -87,6 +90,29 @@ def dyndns_subscribe(subscribe_host="dyndns.yunohost.org", domain=None, key=None
     except requests.ConnectionError:
         raise MoulinetteError(errno.ENETUNREACH, m18n.n('no_internet_connection'))
 
+    # If no password specified in CLI, ask for one to the user
+    if recovery_password is None and msettings.get('interface') == "cli":
```

Comment:
All of those things are actually builtin the moulinette (but sadly! undocumented and not well known :() including password size, we can simplify this code a lot.
https://github.com/YunoHost/yunohost/blob/unstable/data/actionsmap/yunohost.yml#L135-L139